tests/integration/test_charm.py

```diff
@@ -83,13 +84,19 @@ async def test_s3_storage(
     )
     channel = response.json()
 
-    # upload a file
+    # create a test file
+    tmp_path.mkdir(exist_ok=True)
+    test_file = tmp_path / "test_file.txt"
+    test_content = "This is a test file."
+    test_file.write_text(test_content)
```

Comment:
Should this include encoding? I have given that feedback a few times, I'll add an item to my todo list to write a standard for that

Reply:
Fixed